<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">
	<!-- Root Context: defines shared resources visible to all other web components -->
	<!--Configuring DataSource -->

	<bean id="datasource" class="org.apache.commons.dbcp.BasicDataSource">
		<property name="driverClassName">
			<value>com.mysql.jdbc.Driver</value>
		</property>
		<property name="url">
			<value>jdbc:mysql://localhost:3306/employee</value>
		</property>
		<property name="username">
			<value>root</value>
		</property>
		<property name="password">
			<value>root</value>
		</property>
	</bean>

	<!-- Configure JDBC Template -->
	<bean id="jdbctemp" class="org.springframework.jdbc.core.JdbcTemplate">
		<constructor-arg>
			<ref local="datasource" />
		</constructor-arg>
	</bean>
	<bean id="loginModel" class="com.santosh.spring.LoginModel">
		<constructor-arg>
			<ref local="jdbctemp" />
		</constructor-arg>
	</bean>
	<bean id="logincnt" class="com.santosh.spring.LoginController">
		<property name="loginModel">
			<ref local="loginModel" />
		</property>
	</bean>
	<bean id="myurlmapping"
		class="org.springframework.web.servlet.handler.SimpleUrlHandlerMapping">
		<property name="mappings">
			<props>
				<prop key="/views/login.spring">logincnt</prop>
			</props>
		</property>
	</bean>
	<bean id="handlerExceptionResolver"
		class="org.springframework.web.servlet.handler.SimpleMappingExceptionResolver">
		<property name="exceptionMappings">
			<props>
				<prop key="com.santosh.spring.MyException">
					/Error.html
				</prop>
			</props>
		</property>
	</bean>
</beans>
	